crates/c-api/include/wasmtime/engine.h

```diff
@@ -0,0 +1,35 @@
+/**
+ * \file wasmtime/engine.h
+ *
+ * Wasmtime-specific extensions to #wasm_engine_t.
+ */
+
+#ifndef WASMTIME_ENGINE_H
+#define WASMTIME_ENGINE_H
+
+#include <wasm.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \brief Increments the engine-local epoch variable.
+ *
+ * This function will increment the engine's current epoch which can be used to
+ * force WebAssembly code to trap if the current epoch goes beyond the
+ * #wasmtime_store_t configured epoch deadline.
+ *
+ * This method is safe to call from any thread.
```

Comment:
Should we note here that it's also signal-safe (I think we have a similar note in the rustdoc text)?